bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/driver/IOStream.java

```diff
@@ -51,20 +55,12 @@ public void stop() {
      * @param readSize size to read
      * @return number of bytes read
      */
-    public int read(byte[] b, int offset, int readSize) throws InterruptedException {
+    public int read(byte[] b, int offset, int readSize) throws InterruptedException, IOException {
         int len = 0;
         while (!stopped && len < 1) {
-            try {
-                len = in.read(b, offset, readSize);
-                if (Thread.interrupted()) {
-                    throw new InterruptedException();
-                }
-            } catch (IOException e) {
-                logger.trace("got exception while reading: {}", e.getMessage());
-                while (!stopped && !reconnect()) {
-                    logger.trace("sleeping before reconnecting");
-                    Thread.sleep(10000);
-                }
+            len = in.read(b, offset, readSize);
```

Comment:
One thing to note, if `in.read` returns a -1 here, which would indicate the end of the stream, then this code turns into a busy loop.

There are two ways you can fix this:
If the calling code is expected to handle an eof then you could just break out of the loop and return -1 to the caller.
```suggestion
            len = in.read(b, offset, readSize);
            if(len == -1){
                break; 
            }
```
If there is no expectation for the caller to handle a return value of -1, then I suggest throwing an exception instead.
```suggestion
            len = in.read(b, offset, readSize);
            if(len == -1){
                throw new EOFException();
            }
```

Reply:
Nice catch, I'm going to have it throw an EOFException. It's been this way for years.